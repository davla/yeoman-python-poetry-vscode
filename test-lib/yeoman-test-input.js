/* eslint-disable import/prefer-default-export */

export const withInput = (runContext, { optionName, promptName, inputValue }) =>
  Math.random() > 0.5
    ? runContext.withOptions({ [optionName]: inputValue })
    : runContext.withPrompts({ [promptName]: inputValue });
