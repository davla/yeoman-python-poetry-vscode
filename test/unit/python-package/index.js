import path from "node:path";

import yeomanTest from "yeoman-test";

import PythonPackageGenerator from "../../../generators/python-package/index.js";
import restoreRunResult from "../../lib/generator-hooks.js";
import { withInput } from "../../lib/yeoman-test-input.js";

const generatorInput = [
  {
    optionName: "package-name",
    promptName: "packageName",
    inputValue: "tekken_7",
  },
  {
    optionName: "package-version",
    promptName: "packageVersion",
    inputValue: "1.9.0",
  },
];

describe("python-poetry-vscode:python-package", () => {
  beforeEach(function () {
    this.generator = withInput(
      yeomanTest.run(PythonPackageGenerator),
      generatorInput,
    );
  });

  describe("sources", () => {
    afterEach(restoreRunResult);

    it("should create __init__.py in a directory named after the package", async function () {
      this.runResult = await this.generator;
      this.runResult.assertFile(path.join("tekken_7", "__init__.py"));
    });
  });

  describe("tests", () => {
    afterEach(restoreRunResult);

    it("should create __init__.py in the tests directory", async function () {
      this.runResult = await this.generator;
      this.runResult.assertFile(path.join("tests", "__init__.py"));
    });

    it("should create a tests file named after the package in the tests directory", async function () {
      this.runResult = await this.generator;
      this.runResult.assertFile(path.join("tests", "test_tekken_7.py"));
    });
  });
});
