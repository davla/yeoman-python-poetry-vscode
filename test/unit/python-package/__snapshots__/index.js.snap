// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`python-poetry-vscode:python-package sources should populate the pacakge __init__.py file 1`] = `
"__version__ = \\"1.9.0\\"
"
`;

exports[`python-poetry-vscode:python-package tests should populate the test __init__.py file 1`] = `""`;

exports[`python-poetry-vscode:python-package tests should populate the test file 1`] = `
"from package_name import __version__


def test_version():
    assert __version__ == \\"1.9.0\\"
"
`;
